tls-support-plan.md

```diff
@@ -0,0 +1,81 @@
+## TLS Support for Database Handlers
+
+### General approach
++ Extract sslmode params from credentials into backend connection options
+  + sslmode: Provide different levels of protection.
+  + sslca: The Certificate Authority (CA) certificate file. This option, if used, must specify the same certificate used by the server
+  + sslcert: The client public key certificate file.
+  + sslkey: The client private key file.
+  + sslcrl: The certificates revoked by certificate authorities.
++ Dial backend to get net.Conn
++ Default to secure connection if TLS supported
++ Ensure client TLS requirements match server requirements and support, FAIL if not.
+  + Default to insecure connection, regardless of support
+  + Use TLS if server demands it
++ Upgrade net.Conn to tls.Conn and use appropriate strategy for sslmode
+
+### Test cases
+
+Given DB container services with TLS support and without:
++ each sslmode works or errors as intended
```

Comment:
is it worth writing out exactly what this means right now?

Reply:
Stories provide more detail